packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -0,0 +1,257 @@
+  /// ```
+  final String semanticsLabel;
+
+  /// Utility method to convert InlineSpan into [TextSpan] or [WidgetSpan].
+  ///
+  /// Will return null if span is not of type T or null. Otherwise, return
+  /// the span cast as T. Values of T that are not subclasses of [InlineSpan]
+  /// will result in null.
+  static T asType<T>(InlineSpan span) {
+    if (span is! T)
+      return null;
+    return span as T;
+  }
+
+  /// Apply the properties of this object to the given [ParagraphBuilder], from
+  /// which a [Paragraph] can be obtained. [Paragraph] objects can be drawn on
+  /// [Canvas] objects.
+  void build(ui.ParagraphBuilder builder, { double textScaleFactor = 1.0, List<PlaceholderDimensions> dimensions });
+
+  /// Walks this InlineSpan and any descendants in pre-order and calls [visitor]
+  /// for each span that has content.
+  ///
+  /// When [visitor] returns true, the walk will continue. When [visitor] returns
+  /// false, then the walk will end.
+  ///
+  /// The type of the span may be checked using [asType].
+  bool visitInlineSpan(bool visitor(InlineSpan span));
+
+  /// Returns the text span that contains the given position in the text.
+  InlineSpan getSpanForPosition(TextPosition position);
+
+  /// Flattens the [InlineSpan] tree into a single string.
+  ///
+  /// Styles are not honored in this process.
+  String toPlainText();
+
+  /// Returns the UTF-16 code unit at the given index in the flattened string.
+  ///
+  /// Returns null if the index is out of bounds.
+  int codeUnitAt(int index);
+
+  /// In checked mode, throws an exception if the object is not in a
+  /// valid configuration. Otherwise, returns true.
+  ///
+  /// This is intended to be used as follows:
+  ///
+  /// ```dart
+  /// assert(myInlineSpan.debugAssertIsValid());
+  /// ```
+  bool debugAssertIsValid();
+
+  /// Describe the difference between this span and another, in terms of
+  /// how much damage it will make to the rendering. The comparison is deep.
+  ///
+  /// Comparing a [TextSpan] with a [WidgetSpan] will result in [RenderComparison.layout].
+  ///
+  /// See also:
+  ///
+  ///  * [TextStyle.compareTo], which does the same thing for [TextStyle]s.
+  RenderComparison compareTo(InlineSpan other);
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    final InlineSpan typedOther = other;
+    return typedOther.style == style
+        && typedOther.recognizer == recognizer
+        && typedOther.semanticsLabel == semanticsLabel;
+  }
+
+  @override
+  int get hashCode => hashValues(style, recognizer);
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.defaultDiagnosticsTreeStyle = DiagnosticsTreeStyle.whitespace;
+    // Properties on style are added as if they were properties directly on
+    // this InlineSpan.
```

Comment:
this comment doesn't really say anything useful (it just repeats what the code says). (this isn't new in your patch)

Reply:
Fix this one?